framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -2257,19 +2257,33 @@ public ParameterizedExecutableType constructorFromUse(NewClassTree tree) {
         AnnotatedTypeMirror type = fromNewClass(tree);
         addComputedTypeAnnotations(tree, type);
         AnnotatedExecutableType con = getAnnotatedType(ctor); // get unsubstituted type
+
+        if (tree.getClassBody() != null) {
+            // Since anonymous constructor bodies only consist of the super constructor invocation,
```

Comment:
How about: "The artificial constructor for an anonymous class only contains an invocation of the corresponding super class constructor. The artificial constructor does not have all the type annotations the super class constructor has, but conceptually it should have the same type annotations. Instead of copying all the annotations over, we simply use the super constructor element for the anonymous class constructor."